src/test/java/TopNavigationTest.java

```diff
@@ -0,0 +1,123 @@
+import components.TopNavigation;
+import enums.Attribute;
+import org.assertj.core.api.Assertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import utils.WebDriverUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TopNavigationTest {
+    private static final String HOMEPAGE_URL = "https://magento.softwaretestingboard.com/";
+
+    private WebDriver driver;
+    private TopNavigation topNavigation;
+    private List<WebElement> topCategoriesList;
+
+    @BeforeClass
+    public void setUp() {
+        driver = WebDriverUtils.createWebDriver();
+        driver.get(HOMEPAGE_URL);
+        topNavigation = new TopNavigation(driver);
+        topCategoriesList = topNavigation.getTopCategoriesList();
+    }
+
+    @Test
+    public void topNavigation_TopLevelCategories_CorrectlyDisplayed() {
+        //then
+        List<String> expectedNamesList = List.of("What's New", "Women", "Men", "Gear", "Training", "Sale");
+        List<String> expectedLinkList = List.of("what-is-new.html", "women.html", "men.html",
+                "gear.html", "training.html", "sale.html");
+        validateMenuCategoriesNamesAndLinks(topCategoriesList, expectedNamesList, expectedLinkList);
+    }
+
+    @DataProvider
+    public Object[][] secondCategoriesLevelWithExpectedNamesAndLinks() {
+        topCategoriesList = topNavigation.getTopCategoriesList();
```

Comment:
Should be removed. private List<WebElement> topCategoriesList has been already defined as a class field.